tests/test_core.py

```diff
@@ -4932,8 +4932,8 @@ def test_unistd_unlink(self):
       if WINDOWS and fs == 'NODEFS':
         Building.COMPILER_TEST_OPTS += ['-DNO_SYMLINK=1']
       self.do_run(src, 'success', force_c=True, js_engines=[NODE_JS])
-    # Several differences/bugs on Windows including https://github.com/nodejs/node/issues/18014
-    if not WINDOWS:
+    # Several differences/bugs on non-linux including https://github.com/nodejs/node/issues/18014
+    if not WINDOWS and not MACOS:
```

Comment:
I nicer fix here might be to split this test in two and add @no_windows(..bug..) and @no_macos(..bug..).

But either way lgtm.

Reply:
Yeah, sadly though it's not possible to directly use those decorators because they'd disable the whole test, and we want to just disable part of the test.

Reply:
Thats why I suggest splitting the test.  If you are at the point where half of it fails, then its probably a sign that the test too big and needs to be split.  No need to do that now, just a general note.

Reply:
Oh, sorry, I missed that earlier. Yeah, good point, splitting would be nicer in general.